src/post_processing.jl

```diff
@@ -707,3 +690,77 @@ function write_results(
 
     return Dict("bus_results" => bus_df, "flow_results" => branch_df)
 end
+
+"""
+Modify the values in the given `System` to correspond to the given `PowerFlowData` such that
+if a new `PowerFlowData` is constructed from the resulting system it is the same as `data`.
+See also `write_powerflow_solution!`. NOTE that this assumes that `data` was initialized
+from `sys` and then solved with no further modifications.
+"""
+function update_system!(sys::PSY.System, data::PowerFlowData; time_step = 1)
+    for bus in PSY.get_components(PSY.Bus, sys)
+        bus_number = data.bus_lookup[PSY.get_number(bus)]
+        bus_type = data.bus_type[bus_number, time_step]  # use this instead of bus.bustype to account for PV -> PQ
+        if bus_type == PSY.ACBusTypes.REF
+            # For REF bus, voltage and angle are fixed; update active and reactive
+            P_gen = data.bus_activepower_injection[bus_number, time_step]
+            Q_gen = data.bus_reactivepower_injection[bus_number, time_step]
+            _power_redistribution_ref(sys, P_gen, Q_gen, bus,
+                DEFAULT_MAX_REDISTRIBUTION_ITERATIONS)
+        elseif bus_type == PSY.ACBusTypes.PV
+            # For PV bus, active and voltage are fixed; update reactive and angle
+            Q_gen = data.bus_reactivepower_injection[bus_number, time_step]
+            _reactive_power_redistribution_pv(sys, Q_gen, bus,
+                DEFAULT_MAX_REDISTRIBUTION_ITERATIONS)
+            PSY.set_angle!(bus, data.bus_angles[bus_number, time_step])
+        elseif bus_type == PSY.ACBusTypes.PQ
+            # For PQ bus, active and reactive are fixed; update voltage and angle
+            Vm = data.bus_magnitude[bus_number, time_step]
+            PSY.set_magnitude!(bus, Vm)
+            PSY.set_angle!(bus, data.bus_angles[bus_number, time_step])
+            # if it used to be a PV bus, also set the Q value:
+            if bus.bustype == PSY.ACBusTypes.PV
+                Q_gen = data.bus_reactivepower_injection[bus_number, time_step]
+                _reactive_power_redistribution_pv(sys, Q_gen, bus,
+                    DEFAULT_MAX_REDISTRIBUTION_ITERATIONS)
+                # now both the Q and the Vm, Va are correct for this kind of buses
+            end
+        end
+    end
+end
+
+# This cannot work because we do not know the redistribution keys from the power values of devices 
```

Comment:
WE need to document this in an issue to discuss the dependency with the network model